frontend/src/metabase/components/EntityItem.jsx

```diff
@@ -34,22 +35,26 @@ const EntityItem = ({
   onToggleSelected,
   selectable,
   variant,
+  item,
 }) => {
   const actions = [
     onPin && {
       title: t`Pin this item`,
       icon: "pin",
       action: onPin,
+      event: `${analyticsContext};Entity Item;${item.model};Pin Item`,
```

Comment:
@salsakran Any preference on the event structure here? Not sure if it should be `Collection Landing -> Entity Item -> Pin item -> Dashboard` or `Collection Landing -> Entity Item -> Dashbaord -> Pin item`